File: frontend/src/types/history.ts
export interface HistoryItem {
  id: number;
  tool_name: string;
  input_preview: string | null;
  result_preview: string | null;
  created_at: string;
}

export interface HistoryCreatePayload {
  tool_name: string;
  input_preview?: string | null;
  result_preview?: string | null;
}

File: backend/app/models/history.py
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import Base


class HistoryItem(Base):
    __tablename__ = "history_items"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    tool_name: Mapped[str] = mapped_column(String(100), nullable=False)
    input_preview: Mapped[str | None] = mapped_column(Text, nullable=True)
    result_preview: Mapped[str | None] = mapped_column(Text, nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )

    user = relationship("User", back_populates="history_items")
